src/frontend/src/lib/services/rest.services.ts

```diff
@@ -10,35 +8,29 @@ export interface RestRequestParams<Response, Error = unknown, Success = void> {
  * A utility function to make a REST request with possible retries.
  *
  * @param request - The request function to call.
- * @param onSuccess - The function to call when the request is successful.
- * @param onError - The optional function to call when the request fails and the max retries are reached.
  * @param onRetry - The optional function to call when the request fails and a retry is needed.
  * @param maxRetries - The maximum number of retries to attempt.
- * @returns The result of the onSuccess/onError function or undefined if the max retries are reached.
+ * @returns The result of the request or it throws an error if the max retries are reached.
```

Comment:
I think you may be able to document the error with `@throws`

Reply:
done!